### PR TITLE
Remove cyclic references and more layers

### PR DESCRIPTION
**Pull request by DonJayamanne**
After https://github.com/microsoft/vscode-jupyter/pull/14164

### CODE REVIEW

File: src/kernels/jupyter/session/jupyterSessionManager.ts [hunk 1]
```diff
@@ -60,14 +49,22 @@ export class JupyterSessionManager implements IJupyterSessionManager {
         return this._jupyterlab!;
     }
     constructor(
-        _config: IConfigurationService,
         private outputChannel: IOutputChannel,
         private configService: IConfigurationService,
         private readonly kernelService: IJupyterKernelService | undefined,
         private readonly backingFileCreator: IJupyterBackingFileCreator,
         private readonly requestCreator: IJupyterRequestCreator,
-        private readonly jupyterConnection: JupyterConnection
```
Comment (DonJayamanne) [inline src/kernels/jupyter/session/jupyterSessionManager.ts:69]: Cyclic reference removed completely

File: src/kernels/jupyter/session/jupyterSessionManager.ts [hunk 2]
```diff
@@ -76,26 +73,21 @@ export class JupyterSessionManager implements IJupyterSessionManager {
         this.disposed = true;
         traceVerbose(`Disposing session manager`);
         try {
-            if (this.contentsManager) {
```
Comment (DonJayamanne) [inline src/kernels/jupyter/session/jupyterSessionManager.ts:79]: No more nullables.

File: src/kernels/jupyter/session/jupyterSessionManager.ts [hunk 3]
```diff
@@ -104,22 +96,7 @@ export class JupyterSessionManager implements IJupyterSessionManager {
         }
     }
 
-    public async initialize(connInfo: IJupyterConnection): Promise<void> {
```
Comment (DonJayamanne) [inline src/kernels/jupyter/session/jupyterSessionManager.ts:107]: No more sync.

File: src/kernels/jupyter/session/jupyterSessionManagerFactory.ts [hunk 4]
```diff
@@ -24,27 +22,24 @@ export class JupyterSessionManagerFactory implements IOldJupyterSessionManagerFa
...
+        @inject(IAsyncDisposableRegistry) private readonly asyncDisposables: IAsyncDisposableRegistry
     ) {}
 
     /**
      * Creates a new IJupyterSessionManager.
      * @param connInfo - connection information to the server that's already running.
      * @param failOnPassword - whether or not to fail the creation if a password is required.
      */
-    public async create(connInfo: IJupyterConnection): Promise<IJupyterSessionManager> {
+    public create(connInfo: IJupyterConnection): IJupyterSessionManager {
```
Comment (DonJayamanne) [inline src/kernels/jupyter/session/jupyterSessionManagerFactory.ts:33]: No more aync, in another PR this whole class will go away.

**Review by aeschli — APPROVED** [review]

**Pull request merged**